File: hardware_store/internal/model/address/address.go
package address

import (
	"github.com/google/uuid"
)

type Address struct {
	AddressID uuid.UUID `validate:"required"`
	Country   string    `validate:"required,min=2,max=50"`
	City      string    `validate:"required,min=2,max=50"`
	Street    string
}

File: store_python/src/app/dto/address.py
from marshmallow import Schema, fields, validate


class AddressSchema(Schema):
    """DTO: адрес (ответ)."""
    id = fields.UUID(dump_only=True)
    country = fields.Str(required=True, validate=validate.Length(min=1, max=100))
    city = fields.Str(required=True, validate=validate.Length(min=1, max=100))
    street = fields.Str(required=True, validate=validate.Length(min=1, max=255))


class AddressCreateSchema(Schema):
    """DTO: создание/обновление адреса (вход)."""
    country = fields.Str(required=True, validate=validate.Length(min=1, max=100))
    city = fields.Str(required=True, validate=validate.Length(min=1, max=100))
    street = fields.Str(required=True, validate=validate.Length(min=1, max=255))
